### PR TITLE
-added a test to check the error from the issue 1820

### PR DESCRIPTION
**Pull request by Bastczuak**
…happen.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch.
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: test/specs/wrapper/find.spec.js [hunk 1]
```diff
@@ -279,6 +280,68 @@ describeWithShallowAndMount('find', mountingMethod => {
...
+    const routes = [
+      {
+        path: '/a/b',
+        name: 'ab',
+        component: TestComponentToFind
+      }
+    ]
+    const router = new VueRouter({ routes })
+    // and this will not work with shallowMount because router view gets stubbed! How can I just test mount?
+    const wrapper = mountingMethod(
```
Comment (lmiller1990) [inline test/specs/wrapper/find.spec.js]: `mountingMethod` is both `mount` and `shallowMount`. We run all tests for both. If you don't want to run the test for one, just import the one you want on line 2 from `packages/src/test-utils`. Or you can do a check with the `doNotTestIf` helpers.

Comment (Bastczuak) [reply test/specs/wrapper/find.spec.js]: ok done that.

**Comment by lmiller1990** [comment]
Thanks, I'll look at this today!

File: packages/test-utils/src/matches.js [hunk 2]
```diff
@@ -56,7 +56,12 @@ export function matches(node, selector) {
     return element && element.matches && element.matches(selector.value)
   }
 
-  const componentInstance = node[FUNCTIONAL_OPTIONS] || node.child
+  // ignore the functional component if its a RouterView
+  // Find a good explanation comment why tahts the case, please help!
+  const componentInstance =
+    node[FUNCTIONAL_OPTIONS]?.name === 'RouterView'
```
Comment (lmiller1990) [inline packages/test-utils/src/matches.js]: Can we have an extra check? Just to be sure:

`node[FUNCTIONAL_OPTIONS] && node[FUNCTIONAL_OPTIONS].name`

I am not sure if we have `?` enabled in this code base or not.

**Review by lmiller1990 — CHANGES_REQUESTED** [review]
Please add an extra check! thanks.

File: packages/test-utils/src/matches.js (hunk 2 above)
Comment (lmiller1990) [reply packages/test-utils/src/matches.js]: Edit: we do **not** - that's why CI is failing to build. Let's do it old school for now with the `&&` check.

Comment (Bastczuak) [reply packages/test-utils/src/matches.js]: okok

Comment (Bastczuak) [reply packages/test-utils/src/matches.js]: sorry for that, it worked in the tests, that's why I assumed it's generally working.

**Comment by Bastczuak** [comment]
@lmiller1990 ok I added the extra check.

**Comment by Bastczuak** [comment]
@lmiller1990  Dude I am in despair... I can't understand why the karma test are failing and the jest test not. They are almost the same test case. I really really have no idea.

**Comment by lmiller1990** [comment]
The only test failing is the one using the router. This PR makes a change that specifically adds a check against `RouterView` [on this line](https://github.com/vuejs/vue-test-utils/pull/1832/files#diff-80ab213bfb0eeafc02c6f0005a492f52eab27ef052d46ed9e17c5df0da4eefe1R62). I guess we broke something? 

I will try taking a look at this on the weekend. I'd suspect if you comment out that check, the Karma tests will pass. That's the first thing I'm going to do - figure out if this failure is due to a change we made, or not.

**Pull request closed without merging**